### PR TITLE
minijail: Pass all environment variables to the binary

### PR DESCRIPTION
**Pull request by adombeck**
Since we aren't using minijail for security and don't claim that we are using it for hermeticity, we might as well pass all environment variables to the fuzz test. This can help improve compatibility with local dependencies or choices such as `JAVA_HOME`.

### CODE REVIEW

File: pkg/minijail/process_wrapper/src/process_wrapper.c [hunk 1]
```diff
@@ -1,60 +1,27 @@
...
 
-  // Forward all arguments after the separator, which we know exists at this
-  // point.
-  char **executable_argv = argv + separator_index + 1;
-  if (execve(*executable_argv, executable_argv, envp) == -1) {
-    fprintf(stderr, "execve(%s) failed: %s\n", *executable_argv,
-            strerror(errno));
+  // Skip over both the process wrapper's own argv[0], the directory,
+  // and the -- separator.
+  if (execv(argv[3], argv + 3) == -1) {
```
[the remark this answers is not in the document]
Comment (adombeck) [reply pkg/minijail/process_wrapper/src/process_wrapper.c]: done

File: pkg/minijail/process_wrapper/src/process_wrapper.c [hunk 2]
```diff
@@ -1,60 +1,27 @@
...
 #include <string.h>
 #include <unistd.h>
 
-// Includes argv[0] and the working directory.
-#define NUM_FIXED_ARGS 2u
-
-// Executes argv[2] with arguments listed after the separator and the
-// environment variables listed before the separator after changing the working
-// directory to argv[1].
+// Executes argv[3] with argv[4..argc-1] as arguments after changing the
```
[the remark this answers is not in the document]
Comment (adombeck) [reply pkg/minijail/process_wrapper/src/process_wrapper.c]: Updated now that the process wrapper doesn't take a separator anymore

File: util/envutil/envutil.go [hunk 3]
```diff
@@ -103,3 +104,25 @@ func ToMap(env []string) map[string]string {
 	}
 	return res
 }
+
+func QuotedEnv(env []string) []string {
+	var quotedEnv []string
+	for _, e := range env {
+		s := strings.SplitN(e, "=", 2)
+		k, v := s[0], s[1]
+		quotedEnv = append(quotedEnv, fmt.Sprintf("%s=%q", k, v))
```
Comment (fmeum) [inline util/envutil/envutil.go:113]: We could make this `%s='%s'` to appropriately quote `$` in env values, but that's very much an edge case.

**Review by fmeum — APPROVED** [review]

File: pkg/minijail/minijail.go [hunk 4]
```diff
@@ -351,23 +349,23 @@ func NewMinijail(opts *Options) (*minijail, error) {
...
+	args := append(minijailArgs, "--")
+	args = append(args, processWrapperArgs...)
+	args = append(args, opts.Args...)
+
+	// When CI_DEBUG_MINIJAIL is set, we don't execute the actual
+	// libFuzzer command but only print it and start a shell instead.
+	// When used together with SKIP_CLEANUP, this allows to copy the
+	// Minijail command from the logs to open a shell in the sandbox
+	// environment to debug issues interactively.
+	if os.Getenv("CI_DEBUG_MINIJAIL") != "" {
```
Comment (jochil) [inline pkg/minijail/minijail.go:364]: wouldn't this be a good opportunity to also change the `CI` prefix to `CIFUZZ` or even removing it (as in `SKIP_CLEANUP`) ?

File: util/envutil/envutil.go [hunk 5]
```diff
@@ -75,6 +75,20 @@ func Setenv(env []string, key, value string) ([]string, error) {
 	return env, nil
 }
 
+// Copy copies all environment variables from src to dst. When an
+// environment variable is already set in dst, the value in dst is
+// overwritten.
```
Comment (jochil) [inline util/envutil/envutil.go:81]: what's about covering this behavior with a little unit test ;)

File: util/envutil/envutil.go (hunk 3 above)
Comment (adombeck) [reply util/envutil/envutil.go]: done

File: pkg/minijail/minijail.go (hunk 4 above)
Comment (adombeck) [reply pkg/minijail/minijail.go]: I removed the prefix

File: util/envutil/envutil.go (hunk 5 above)
Comment (adombeck) [reply util/envutil/envutil.go:81]: done

**Review by jochil — APPROVED** [review]

**Pull request merged**